logzio-sender/src/main/java/io/logz/sender/LogzioSender.java

```diff
@@ -245,4 +220,85 @@ private void debug(String message, Throwable e) {
         }
     }
 
+
+    public static class Builder {
+        private boolean debug = false;
+        private int drainTimeout = 5; //sec
+        private SenderStatusReporter reporter;
+        private ScheduledExecutorService tasksExecutor;
+        private InMemoryLogsBuffer.Builder InMemoryLogsBufferBuilder;
+        private DiskQueue.Builder diskQueueBuilder;
+        private HttpsRequestConfiguration httpsRequestConfiguration;
+
+        public Builder setDrainTimeout(int drainTimeout) {
+            this.drainTimeout = drainTimeout;
+            return this;
+        }
+
+        public Builder setDebug(boolean debug) {
+            this.debug = debug;
+            return this;
+        }
+
+        public Builder setTasksExecutor(ScheduledExecutorService tasksExecutor) {
+            this.tasksExecutor = tasksExecutor;
+            return this;
+        }
+
+
+        public Builder setReporter(SenderStatusReporter reporter) {
+            this.reporter = reporter;
+            return this;
+        }
+
+        public Builder setHttpsRequestConfiguration(HttpsRequestConfiguration httpsRequestConfiguration) {
+            this.httpsRequestConfiguration = httpsRequestConfiguration;
+            return this;
+        }
+
+        public InMemoryLogsBuffer.Builder WithInMemoryLogsBuffer() {
+            return InMemoryLogsBuffer.builder(this);
+        }
+
+        public DiskQueue.Builder WithDiskMemoryQueue() {
+            return DiskQueue.builder(this, tasksExecutor);
+        }
+
+        void setDiskQueueBuilder(DiskQueue.Builder diskQueueBuilder) {
+            this.diskQueueBuilder = diskQueueBuilder;
+        }
+
+        void setInMemoryLogsBufferBuilder(InMemoryLogsBuffer.Builder InMemoryLogsBufferBuilder) {
+            this.InMemoryLogsBufferBuilder = InMemoryLogsBufferBuilder;
+        }
+
+        public LogzioSender build() throws LogzioParameterErrorException {
+            return  getLogzioSender(
+                    httpsRequestConfiguration,
+                    drainTimeout,
+                    debug,
+                    reporter,
+                    tasksExecutor,
+                    getLogsBuffer()
+            );
+        }
+
+        private LogzioLogsBufferInterface getLogsBuffer() throws LogzioParameterErrorException {
+            LogzioLogsBufferInterface logsBuffer = null;
+            if (diskQueueBuilder != null) {
+                diskQueueBuilder.setDiskSpaceTasks(tasksExecutor);
+                diskQueueBuilder.setReporter(reporter);
+                logsBuffer = diskQueueBuilder.build();
+            }else if (InMemoryLogsBufferBuilder != null) {
+                InMemoryLogsBufferBuilder.setReporter(reporter);
```

Comment:
if `InMemoryLogsBufferBuilder` is a variable should start with small letter. Doesn't IDEA warns on this?